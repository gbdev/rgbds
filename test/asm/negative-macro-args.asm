MACRO mac
	for i, -1, -_NARG - 1, -1
		println "{d:i}: \<i> == \<{d:i}>"
	endr
	; error cases
	def i = 0
	println "{d:i}: \<i> == \<{d:i}>"
	def i = -_NARG - 1
	println "{d:i}: \<i> == \<{d:i}>"
	def i = $7fff_ffff
	println "{d:i}: \<i> == \<{d:i}>"
	; signed/unsigned difference error cases
	def i = $8000_0000
	println "{d:i}: \<i> == \<{d:i}>"
	println "{u:i}: \<i> == \<{u:i}>"
	def i = $ffff_ffff
	println "{d:i}: \<i> == \<{d:i}>"
	println "{u:i}: \<i> == \<{u:i}>"
ENDM

mac A, B, C, D, E, F, G
