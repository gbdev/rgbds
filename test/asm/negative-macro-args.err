error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(7):
    Invalid bracketed macro argument '\<0>'
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(7):
    Invalid bracketed macro argument '\<0>'
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(9):
    Macro argument '\<-8>' not defined
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(9):
    Macro argument '\<-8>' not defined
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(11):
    Macro argument '\<2147483647>' not defined
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(11):
    Macro argument '\<2147483647>' not defined
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(14):
    Macro argument '\<-2147483648>' not defined
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(14):
    Number in bracketed macro argument is too large
while expanding symbol "-2147483648"
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(15):
    Macro argument '\<-2147483648>' not defined
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(15):
    Number in bracketed macro argument is too large
while expanding symbol "2147483648"
error: negative-macro-args.asm(21) -> negative-macro-args.asm::mac(18):
    Number in bracketed macro argument is too large
while expanding symbol "4294967295"
error: Assembly aborted (11 errors)!
