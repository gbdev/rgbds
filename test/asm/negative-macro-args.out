-1: G == G
-2: F == F
-3: E == E
-4: D == D
-5: C == C
-6: B == B
-7: A == A
0:  == 
-8:  == 
2147483647:  == 
-2147483648:  == >
2147483648:  == >
-1: G == G
4294967295: G == >
